content/articles/functional-programming-with-java-lambda/index.md

```diff
@@ -0,0 +1,333 @@
+```
+
+From the first line of the code above is an interface with the name Hello. Which declare a method called `sayHello`, the method takes one parameter of `greet`. Which is of type string and will return type string. The fourth line creates a new class called `LambdaAlgorithmTest` which contains the main method. The method declares a function `hello` with the type `Hello` interface. And the method function has one parameter of `(message)`with two-variable `str1` and `str2` which both are of type string. The first variable `str1` takes a value `Welcome`. And the second variable `str2` concatenates `str1` and `message` together. And the function return `str2`.
+
+### Working with different Lambda parameters
+#### Lambda example that takes no parameters
+
+```java
+
+@FunctionalInterface
+
+interface MyFunctionalInterface {
+
+ //A method with no parameter
+
+ String sayHello();
+
+}
+
+public class Example {
+
+ public static void main(String args[]) {
+
+ // lambda expression
+
+ MyFunctionalInterface msg = () -> "Hello";
+
+ System.out.println(msg.sayHello());
+
+ }
+
+}
+
+```
+
+From the first line of the above code. `@FunctionalInterface` is an annotation ensuring functional interface does not have more than one abstract method. the second line is an interface with the name ` MyFunctionalInterface`, which declares a method called `sayHello`, the method takes no parameter. the method will return a type of string. the third line creates a new class called `Example` which contains the main method. The method declares a function `msg` with type `MyFunctionalInterface`interface. And the function result will print `Hello`.
+
+#### Lambda example with a single parameter
+
+```java
+
+@FunctionalInterface
+
+interface MyFunctionalInterface {
+
+ //A method with a single parameter
+
+ int incrementByFive(int a);
+
+}
+
+public class LambdaSingleParam {
+
+ public static void main(String args[]) {
+
+ // lambda expression with single parameter num
+
+ MyFunctionalInterface f = (num) -> num+5;
+
+ System.out.println(f.incrementByFive(22));
+
+ }
+
+}
+
+```
+
+#### Lambda example with many parameters
+
+```java
+
+interface StringConcat {
+
+ String strConcat(String a, String b);
+
+}
+
+public class LambdaMultipleParameter {
+
+ public static void main(String args[]) {
+
+ // lambda expression with many arguments
+
+ StringConcat s = (str1, str2) -> str1 + str2;
+
+ System.out.println("Result: "+ s.sconcat("Hello ", "World"));
+
+ }
+
+}
+
+```
+
+### Lambda stream pipeline
+Streams are objects of classes that implement interface Stream. the stream is either from or one of the specific streams interfaces for preparing collections of int, long or double values. Stream enables you to act on the collection of elements with lambda.
+
+Lambda Stream pipeline is a sequence of processing steps from the data source (from collections). The pipeline performs two main operations which is an intermediate or terminal operation. Which formed a chain method calls.
```

Comment:
Check on the two last sentences here and join them so that they can make because they don't.